### PR TITLE
Test profile on checkout step

### PR DESCRIPTION
**Pull request by wdsrocha**
#### What does this PR do?

Tests the flow to successfully fill the profile during the checkout step. As this is the first functional test, it also intends to serve as a base for future tests.

Few observations:
- I tried to keep things simple, so for example, the test file isn't on a specific directory or have a non-generic name. I believe this should change only when needed (i.e.: when new tests are created)
- Things like my personal workspace being used on `cypress.json` and `vtex local token` being passed directly to `yarn e2e` script are temporary and will change when integrating with [tester-hub](https://www.github.com/vtex/tester-hub).
- ~Until [this](https://www.github.com/vtex-apps/checkout-profile/pull/5) isn't merged, the test will fail on the last step (profile to shipping transition).~ (the branch was merged, the tests should pass)

#### How to test it?

Clone it locally and run `yarn e2e`. Don't forget to login on VTEX (toolbelt).

#### Describe alternatives you've considered, if any.

There are many things that could be improved or made differently, but I don't think it really matters that much as this is just the beginning of v0 e2e tests and tester-hub isn't even integrated yet.

### CODE REVIEW

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  cy.visit(`/cart/add?${values}`)
+})
+
+Cypress.Commands.add('fillEmail', () => {
+  cy.get('input').type(`${getRandomEmail()}{enter}`)
+})
+
+Cypress.Commands.add('openCheckoutProfileStep', () => {
+  cy.get(
+    `:nth-child(1) > .vtex-checkout-step-group-0-x-stepTitle > .pl4 > .vtex-button`
```
Comment (lucasecdb) [inline cypress/support/commands.ts:67]: I think we can add some test-ids to those buttons, what do you think?

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  ).click()
+})
+
+Cypress.Commands.add('fillProfile', () => {
+  cy.get('[name=firstName]').type('John')
+  cy.get('[name=lastName]').type('Doe')
+  cy.get('[name=phone]').type('21999999999')
+  cy.get('[name=document]').type(getDocument())
+  cy.get('form.mt6 > .vtex-button').click()
```
Comment (lucasecdb) [inline cypress/support/commands.ts:76]: same for here

File: cypress/support/commands.ts (hunk 1 above)
Comment (wdsrocha) [reply cypress/support/commands.ts:67]: Sounds great. I'll open a PR in the other repo then.

File: cypress.json [hunk 3]
```diff
@@ -1 +1,9 @@
-{}
+{
+  "baseUrl": "https://wdsrocha--checkoutio.myvtex.com/",
```
Comment (jeffersontuc) [inline cypress.json:2]: Is `baseUrl` used? If so, is this hardcoded workspace here intentional?

Comment (wdsrocha) [reply cypress.json:2]: Yes and yes. But I just realized that I can generalise this to `https://<workspace>--<account>.myvtex.com/` (this is required by `tester-hub`) and override it locally using `yarn e2e --baseUrl "https://wdsrocha--checkoutio.myvtex.com/"`. Will update it soon.

**Comment by wdsrocha** [comment]
@lucasecdb, IMO we can merge it before integrating with tester-hub, so at least local tests work.
But I'll still open a PR in `checkout-profile` and `checkout-step-group` to update the testIds.

File: cypress/support/commands.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  cy.get('input').type(`${getRandomEmail()}{enter}`)
+})
+
+Cypress.Commands.add('fillProfile', () => {
+  cy.getByTestId('profile-first-name-wrapper').type('John')
+  cy.getByTestId('profile-last-name-wrapper').type('Doe')
+  cy.getByTestId('profile-phone-wrapper').type('21999999999')
+  cy.getByTestId('profile-document-wrapper').type(getDocument())
+  cy.get('[id=save-personal-info]').check()
+  cy.get('[id=optin-newsletter]').check()
```
Comment (marcoskwkm) [inline cypress/support/commands.ts:71]: Is there any reason to use `[id=foo]` instead of `#foo` here?

Comment (wdsrocha) [reply cypress/support/commands.ts:71]: Oh, I forgot about that! Will update.

**Pull request merged**